docs/source/yosys_internals/hashing.rst

```diff
@@ -0,0 +1,95 @@
+Hashing and associative data structures in Yosys
+------------------------------------------------
+
+Yosys heavily relies on custom data structures such as dict or pool defined in
+kernel/hashlib.h. There are various reasons for this.
+
+The hash function
+~~~~~~~~~~~~~~~~~
+
+The hash function generally used in Yosys is the XOR version of DJB2:
+
+``state = ((state << 5) + state) ^ value``
```

Comment:
If you're just wanting this line of code to be on it's own I would recommend changing it to a(n anonymous) code block instead (the suggestion), otherwise drop the new lines to inline it.

```suggestion
``::

   state = ((state << 5) + state) ^ value
```